### PR TITLE
[SPARK-27492][DOC][FOLLOWUP] Update resource scheduling user docs

### PR DESCRIPTION
**Pull request by jiangxb1987**
### What changes were proposed in this pull request?

Fix a config name typo from the resource scheduling user docs. In case users might get confused with the wrong config name, we'd better fix this typo.

### How was this patch tested?

Document change, no need to run test.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -2639,7 +2639,7 @@ Also, you can modify or add configurations at runtime:
...
+There are configurations available to request resources for the driver: <code>spark.driver.resource.{resourceName}.amount</code>, request resources for the executor(s): <code>spark.executor.resource.{resourceName}.amount</code> and specify the requirements for each task: <code>spark.task.resource.{resourceName}.amount</code>. The <code>spark.driver.resource.{resourceName}.discoveryScript</code> config is required on YARN, Kubernetes and a client side Driver on Spark Standalone. <code>spark.executor.resource.{resourceName}.discoveryScript</code> config is required for YARN and Kubernetes. Kubernetes also requires <code>spark.driver.resource.{resourceName}.vendor</code> and/or <code>spark.executor.resource.{resourceName}.vendor</code>. See the config descriptions above for more information on each.
```
Comment (jiangxb1987) [inline docs/configuration.md]: `spark.driver.executor.{resourceName}.discoveryScript` -> `spark.executor.resource.{resourceName}.discoveryScript`

**Comment by jiangxb1987** [comment]
cc @tgravescs

**Review by tgravescs — APPROVED** [review]
thanks for fixing this

**Comment by jiangxb1987** [comment]
Thanks @tgravescs @dongjoon-hyun , merging to master!

**Pull request closed without merging**